src/poetry/mixology/version_solver.py

```diff
@@ -487,18 +483,21 @@ def _add_incompatibility(self, incompatibility: Incompatibility) -> None:
                 incompatibility
             )
 
-    def _get_locked(self, dependency: Dependency) -> Package | None:
+    def _get_locked(
+        self, dependency: Dependency, *, allow_similar: bool = False
+    ) -> DependencyPackage | None:
         if dependency.name in self._use_latest:
             return None
 
-        locked = self._locked.get(dependency.name)
-        if not locked:
-            return None
-
-        if not dependency.is_same_package_as(locked):
-            return None
-
-        return locked
+        locked = self._locked.get(dependency.name, [])
+        for package in locked:
+            if (allow_similar or dependency.is_same_package_as(package)) and (
```

Comment:
@radoering can you also push the `allow_similar` flag to poetry core as well please?

Reply:
python-poetry/poetry-core#321